exonum-java-binding-core/rust/src/utils/errors.rs

```diff
@@ -46,6 +48,35 @@ pub fn check_error_on_exception<T>(env: &JNIEnv, result: JniResult<T>) -> Result
     })
 }
 
+/// Handles exceptions after executing transactions
+///
+/// The TransactionExecutionException and its descendants are converted into `Error`s with their
+/// descriptions. The rest (Java and JNI errors) are treated as unrecoverable and result in a panic.
+///
+/// Panics:
+/// - Panics if there is some JNI error.
+/// - If there is a pending Java throwable that IS NOT an instance or subclass of the
+/// `TransactionExecutionException`.
+pub fn check_transaction_execution_result<T>(
```

Comment:
I don't think this class belongs here — (in errors) — it is used in TransactionProxy only and must be a private method there (not a public in separate module) and tested through the `Transaction` public interface.

It must also include the error code from the exception — that’s the whole point of this exception type.